src/components/node-list/index.js

```diff
@@ -1,40 +1,247 @@
+    });
+
+    if (isTagType(type.id)) {
+      Object.assign(group, {
+        kind: 'filter',
+        checked: !group.allUnset,
+        visibleIcon: group.allChecked ? 'indicator' : 'indicatorPartial',
+        invisibleIcon: 'indicatorOff'
+      });
+    }
+
+    return groups;
+  }, {});
+
+  const onItemClick = item => {
+    if (isTagType(item.type)) {
+      onTagChange(item, item.checked);
+    } else {
+      if (item.disabled || nodeSelected[item.id]) {
+        onToggleNodeClicked(null);
+      } else {
+        onToggleNodeClicked(item.id);
+      }
+    }
+  };
+
+  const onItemChange = (item, checked) => {
+    if (isTagType(item.type)) {
+      onTagChange(item, checked);
+    } else {
+      if (checked) {
+        onToggleNodeHovered(null);
+      }
+
+      onToggleNodesDisabled([item.id], checked);
+    }
+  };
+
+  const onItemMouseEnter = item => {
+    if (isTagType(item.type)) {
+      onToggleTagActive(item.id, true);
+    } else {
+      if (item.visible) {
+        onToggleNodeHovered(item.id);
+      }
+    }
+  };
+
+  const onItemMouseLeave = item => {
+    if (isTagType(item.type)) {
+      onToggleTagActive(item.id, false);
+    } else {
+      if (item.visible) {
+        onToggleNodeHovered(null);
+      }
+    }
+  };
+
+  const onToggleGroupChecked = (type, checked) => {
+    if (isTagType(type)) {
+      const itemsOfType = items[type] || [];
+      const groupAllUnset = itemsOfType.every(item => item.unset);
+      const allTagsValue = groupAllUnset ? true : undefined;
+      itemsOfType.forEach(tag => onToggleTagFilter(tag.id, allTagsValue));
+    } else {
+      onToggleTypeDisabled(type, checked);
+    }
+  };
+
+  const onTagChange = (tag, checked) => {
+    const valuesAfter = Object.values({ ...tagsEnabled, [tag.id]: !checked });
+    const allDisabled = valuesAfter.every(
+      enabled => typeof enabled === 'undefined' || enabled === false
+    );
+
+    if (allDisabled) {
+      tags.forEach(tag => onToggleTagFilter(tag.id, undefined));
+    } else {
+      onToggleTagFilter(tag.id, !checked);
+    }
+  };
+
+  // Deselect node on Escape key
+  const handleKeyDown = event => {
+    utils.handleKeyEvent(event.keyCode, {
+      escape: () => onToggleNodeClicked(null)
+    });
+  };
+  useEffect(() => {
+    window.addEventListener('keydown', handleKeyDown);
+    return () => window.removeEventListener('keydown', handleKeyDown);
+  });
 
   return (
-    <React.Fragment>
-      <NodeListSearch
-        onUpdateSearchValue={updateSearchValue}
-        searchValue={searchValue}
-      />
-      <Scrollbars
-        className="pipeline-nodelist-scrollbars"
-        style={{ width: 'auto' }}
-        autoHide
-        hideTracksWhenNotNeeded>
-        <NodeListToggleAll nodeIDs={nodeIDs} />
-        <NodeListGroups nodes={filteredNodes} />
-      </Scrollbars>
-    </React.Fragment>
+    <NodeList
+      items={items}
+      sections={sections}
+      groups={groups}
```

Comment:
I know what you mean, I've added a description to this component now which hopefully should help explain how these are being used. I've tried to keep this fairly generic though so that future sections can easily be added to the sidebar.